crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs

```diff
@@ -0,0 +1,193 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use swc_core::ecma::ast::Program;
+use turbo_tasks::primitives::StringVc;
+use turbo_tasks_fs::FileSystemPathVc;
+use turbopack_core::issue::{Issue, IssueSeverity, IssueSeverityVc, IssueVc};
+use turbopack_ecmascript::{CustomTransformer, TransformContext};
+
+#[turbo_tasks::value(transparent)]
+pub struct PluginModule(
+    #[turbo_tasks(trace_ignore)]
+    #[cfg(feature = "swc_ecma_transform_plugin")]
+    pub swc_core::plugin_runner::plugin_module_bytes::CompiledPluginModuleBytes,
+    // Dummy field to avoid turbo_tasks macro complains about empty struct.
+    // This is due to we can't import CompiledPluginModuleBytes by default, it should be only
+    // available for the target / platforms can support swc plugins (which can build wasmer)
+    #[cfg(not(feature = "swc_ecma_transform_plugin"))] pub Option<()>,
+);
+
```

Comment:
I would expect a turbo_tasks::function `CompiledSwcPluginModuleVc::compile(FileSystemPathVc)` constructor here, which allows to compile a WASM file into a compiled module. This would utilize turbo tasks caching and invalidation logic, so updating the WASM will recompile the files...

Reply:
`CompiledPluginModuleBytes` is a struct points to compiled / serialized swc plugin. Compilation itself would occur when read wasm files, and constructing it into CompiledPluginModuleBytes itself immediately. My thought was if a module is created by `PluginModuleVc`, it is managed by turbo which corresponds to the specific compiled binary.

Reply:
Thinking this bit more again, probably we could expose `new` to construct plugin module still.